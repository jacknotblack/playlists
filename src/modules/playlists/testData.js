export const playLists = [
  {
    id: 0,
    name: "aaa",
    songs: [0, 3]
  },
  {
    id: 1,
    name: "bbb",
    songs: []
  },
  {
    id: 2,
    name: "ccc",
    songs: [2]
  },
  {
    id: 3,
    name: "ddd",
    songs: [3]
  },
  {
    id: 4,
    name: "zzz",
    songs: []
  },
  {
    id: 5,
    name: "zzzz",
    songs: []
  },
  {
    id: 6,
    name: "abc",
    songs: []
  },
  {
    id: 7,
    name: "abczzz",
    songs: []
  },
  {
    id: 8,
    name: "123",
    songs: []
  }
];

export const songs = [
  { id: 0, title: "abc1", artist: "Jack", duration: "1:28" },
  { id: 1, title: "abc2", artist: "Tyler", duration: "1:23" },
  { id: 2, title: "abc3", artist: "Tyler", duration: "2:33" },
  { id: 3, title: "abc4", artist: "Randy", duration: "5:41" },
  { id: 4, title: "abc5", artist: "Tyler", duration: "1:34" },
  { id: 5, title: "abc6", artist: "Jack", duration: "4:13" },
  { id: 6, title: "abc7", artist: "Jack", duration: "3:25" },
  { id: 7, title: "abc8", artist: "Randy", duration: "2:53" },
  { id: 8, title: "abc9", artist: "Jack", duration: "2:27" },
  { id: 9, title: "abc10", artist: "Randy", duration: "4:23" },
  { id: 10, title: "abc11", artist: "Tyler", duration: "3:33" }
];
