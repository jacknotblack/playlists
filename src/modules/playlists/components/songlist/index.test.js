import React from "react";
import { shallow } from "enzyme";
import SongList from "./index";
import AddSongPopup from "../../containers/add-song-popup";
import { songs } from "../../testData";

const setup = props => {
  const enzymeWrapper = shallow(<SongList {...props} />);
  return {
    props,
    enzymeWrapper
  };
};

describe("<SongList />", () => {
  const initProps = {
    songs
  };
  describe("render", () => {
    const { enzymeWrapper } = setup(initProps);

    it("renders without crashing", () => {
      expect(enzymeWrapper.length).toBe(1);
    });
    describe("render songList", () => {
      describe("songs length = 0", () => {
        it("should match snapshot", () => {
          const { enzymeWrapper } = setup({
            ...initProps,
            songs: []
          });
          expect(enzymeWrapper).toMatchSnapshot();
        });
      });

      describe("songs is undefined", () => {
        const { enzymeWrapper } = setup({ ...initProps, songs: undefined });
        it("should match snapshot", () => {
          expect(enzymeWrapper).toMatchSnapshot();
        });
      });
    });
  });

  describe("function props", () => {
    const { enzymeWrapper, props } = setup(initProps);

    it("should setState when add to songlist button is clicked", () => {
      enzymeWrapper
        .find(".add")
        .get(0)
        .props.onClick();
      expect(enzymeWrapper.state()).toEqual({
        displayPopop: true,
        songID: 0
      });
      expect(enzymeWrapper).toMatchSnapshot();
    });

    it("should setState when AddSong popup execute onClosing", () => {
      enzymeWrapper
        .find(AddSongPopup)
        .get(0)
        .props.onClosing();
      expect(enzymeWrapper.state()).toEqual({
        displayPopop: false,
        songID: null
      });
    });
  });
});
