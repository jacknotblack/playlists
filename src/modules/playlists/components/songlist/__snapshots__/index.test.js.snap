// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongList /> function props should setState when add to songlist button is clicked 1`] = `
<div
  className="song-list"
>
  <Connect(AddSongPopup)
    onClosing={[Function]}
    songID={0}
  />
  <table>
    <thead>
      <tr>
        <td>
          id
        </td>
        <td>
          Title
        </td>
        <td>
          Artist
        </td>
        <td>
          Duration
        </td>
        <td>
          Add
        </td>
      </tr>
    </thead>
    <tbody>
      <tr
        className="song"
        key="0"
      >
        <td>
          0
        </td>
        <td>
          abc1
        </td>
        <td>
          Jack
        </td>
        <td>
          1:28
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          abc2
        </td>
        <td>
          Tyler
        </td>
        <td>
          1:23
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          abc3
        </td>
        <td>
          Tyler
        </td>
        <td>
          2:33
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="3"
      >
        <td>
          3
        </td>
        <td>
          abc4
        </td>
        <td>
          Randy
        </td>
        <td>
          5:41
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="4"
      >
        <td>
          4
        </td>
        <td>
          abc5
        </td>
        <td>
          Tyler
        </td>
        <td>
          1:34
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="5"
      >
        <td>
          5
        </td>
        <td>
          abc6
        </td>
        <td>
          Jack
        </td>
        <td>
          4:13
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="6"
      >
        <td>
          6
        </td>
        <td>
          abc7
        </td>
        <td>
          Jack
        </td>
        <td>
          3:25
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="7"
      >
        <td>
          7
        </td>
        <td>
          abc8
        </td>
        <td>
          Randy
        </td>
        <td>
          2:53
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="8"
      >
        <td>
          8
        </td>
        <td>
          abc9
        </td>
        <td>
          Jack
        </td>
        <td>
          2:27
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="9"
      >
        <td>
          9
        </td>
        <td>
          abc10
        </td>
        <td>
          Randy
        </td>
        <td>
          4:23
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr
        className="song"
        key="10"
      >
        <td>
          10
        </td>
        <td>
          abc11
        </td>
        <td>
          Tyler
        </td>
        <td>
          3:33
        </td>
        <td
          className="add"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<SongList /> render render songList songs is undefined should match snapshot 1`] = `
<div
  className="song-list"
>
  <span>
    no songs yet
  </span>
</div>
`;

exports[`<SongList /> render render songList songs length = 0 should match snapshot 1`] = `
<div
  className="song-list"
>
  <span>
    no songs yet
  </span>
</div>
`;
